tdrs-backend/tdpservice/parsers/test/test_parse.py

```diff
@@ -1726,3 +1726,291 @@ def test_parse_tribal_section_4_bad_quarter(tribal_section_4_bad_quarter, dfs):
+    m4_cat2_country_fips_code_zero_pad_file.year = 2019
+    m4_cat2_country_fips_code_zero_pad_file.quarter = 'Q1'
+    dfs.save()
+
+    parse.parse_datafile(m4_cat2_country_fips_code_zero_pad_file, dfs)
+
+    parser_errors = ParserError.objects.filter(file=m4_cat2_country_fips_code_zero_pad_file).order_by("pk")
+
+    assert parser_errors.count() == 0
+
+@pytest.fixture
+def tribal_t1_cat2_country_fips_code_zero_pad_file():
+    """Fixture for Tribal T1 COUNTRY_FIPS_CODE without leading zeros."""
+    parsing_file = ParsingFileFactory(
+        year=2024,
+        quarter='Q1',
+        section='Tribal Active Case Data',
+        file__name='tribal_t1_cat2_country_fips_code_zero_pad_file.txt',
+        file__data=(b'HEADER20214A00142TAN1ED\n'
+                    b'T120211011111111119255 140245112 23322   03   0   0   0 502  9   0 0  0   0  0   0  0   0  3   02'
+                    b'22222   0   0222  12     \n'
+                    b'TRAILER0000001         ')
+    )
+    return parsing_file
+
+@pytest.mark.django_db()
+def test_parse_tribal_t1_cat2_country_fips_code_zero_pad_file(tribal_t1_cat2_country_fips_code_zero_pad_file, dfs):
+    """Test parsing a Tribal T1 COUNTRY_FIPS_CODE without leading zeros."""
+    dfs.datafile = tribal_t1_cat2_country_fips_code_zero_pad_file
+    tribal_t1_cat2_country_fips_code_zero_pad_file.year = 2022
+    tribal_t1_cat2_country_fips_code_zero_pad_file.quarter = 'Q1'
+    dfs.save()
+
+    parse.parse_datafile(tribal_t1_cat2_country_fips_code_zero_pad_file, dfs)
+
+    parser_errors = ParserError.objects.filter(file=tribal_t1_cat2_country_fips_code_zero_pad_file).order_by("pk")
+
+    assert parser_errors.count() == 0
+
+@pytest.fixture
+def tribal_t4_cat2_country_fips_code_zero_pad_file():
+    """Fixture for Tribal T4 COUNTRY_FIPS_CODE without leading zeros."""
+    parsing_file = ParsingFileFactory(
+        year=2024,
+        quarter='Q1',
+        section='Tribal Closed Case Data',
+        file__name='tribal_t4_cat2_country_fips_code_zero_pad_file.txt',
+        file__data=(b'HEADER20194C00142TAN1ED\n'
+                    b'T420191011111111762  1 0402451153123                                   \n'
+                    b'TRAILER0000001         ')
+    )
+    return parsing_file
+
+@pytest.mark.django_db()
+def test_parse_tribal_t4_cat2_country_fips_code_zero_pad_file(tribal_t4_cat2_country_fips_code_zero_pad_file, dfs):
+    """Test parsing a Tribal T4 COUNTRY_FIPS_CODE without leading zeros."""
+    dfs.datafile = tribal_t4_cat2_country_fips_code_zero_pad_file
+    tribal_t4_cat2_country_fips_code_zero_pad_file.year = 2020
+    tribal_t4_cat2_country_fips_code_zero_pad_file.quarter = 'Q1'
+    dfs.save()
+
+    parse.parse_datafile(tribal_t4_cat2_country_fips_code_zero_pad_file, dfs)
+
+    parser_errors = ParserError.objects.filter(file=tribal_t4_cat2_country_fips_code_zero_pad_file).order_by("pk")
+
+    assert parser_errors.count() == 0
```

Comment:
since this test performs a transform on the input, can we test the expected value of the field that should be modified?

Reply:
covered this with the paramterized test for the transform itself referenced in the following comment